Index: configs/constants.properties
===================================================================
--- configs/constants.properties	(revision 9876)
+++ configs/constants.properties	(working copy)
@@ -1,7 +1,7 @@
 password=ihasjkIjkal983&pmqekeh6kq[
 mapping_path=../configs/credentials-mapping.xml
-config_path=../../../configs/credentials.xml
-#config_path=../../mvcs-sca/configs/credentials.xml
+#config_path=../../../configs/credentials.xml
+config_path=../../mvcs-sca/configs/credentials.xml
 schema_path=../configs/credentials.xsd
 
 app_title=MVCS Credentials
Index: src/main/java/mil/navy/lcs/mvcs/credentials/server/security/SecurityManager.java
===================================================================
--- src/main/java/mil/navy/lcs/mvcs/credentials/server/security/SecurityManager.java	(revision 9876)
+++ src/main/java/mil/navy/lcs/mvcs/credentials/server/security/SecurityManager.java	(working copy)
@@ -35,7 +35,7 @@
    private static final Logger logger = Logger.getLogger(SecurityManager.class);
 
    /** Change this value when switching between testing and production environments. **/
-   private boolean testing = false;
+   private boolean testing = true;
 
    private static SecurityManager smInstance;
    private final SecurityWorker sw;
@@ -126,6 +126,14 @@
             sp.setAuthenticated(true);
             sp.setUserType(Constants.getConstant(ConstantsEnum.UNIX_GROUP_MAINTENANCE));
          }
+         // username is empty
+         else
+         {
+            sp.setAuthenticated(true);
+            userName = Constants.getConstant(ConstantsEnum.UNIX_GROUP_OPERATOR);
+            sp.setUserName(userName);
+            sp.setUserType(userName);
+         }
 
          return sp;
       }
